### PR TITLE
add docs for using SA as oauth client

### PR DESCRIPTION
**Pull request by deads2k**
Adds docs for https://trello.com/c/DKVf6jJF .

@liggitt for accuracy.

### CODE REVIEW

File: architecture/additional_concepts/authentication.adoc [hunk 1]
```diff
@@ -176,6 +176,20 @@ $ oc create -f <(echo '
 <3> The `redirect_uri` parameter specified in requests to `_<master>_/oauth/authorize` and `_<master>_/oauth/token` must be equal to (or prefixed by) one of the URIs in `redirectURIs`.
 ====
 
+*Service Accounts as OAuth Clients*
+
+A service account can be used as a very constrained kind of OAuth client without any
+cluster-admin intervension.  Service accounts can only request a subset of scopes that allow
```
Comment (liggitt) [inline architecture/additional_concepts/authentication.adoc]: drop "without any cluster-admin intervension"?

**Comment by liggitt** [comment]
looks fine, @adellape for placeholder style

**Comment by adellape** [comment]
@deads2k @liggitt LGTM, labeling for 3.3.

**Comment by adellape** [comment]
@bfallonf Looks like you already followed up on this one. Cool to remove the `to_followup` label here?

**Pull request merged**